sql/modules/Reconciliation.sql

```diff
@@ -207,18 +231,24 @@ CREATE OR REPLACE FUNCTION reconciliation__report_approve (in_report_id INT) ret
                 LEFT JOIN cr_report_line rl ON (rl.report_id = in_report_id
                         AND ((rl.ledger_id = ac.entry_id
                                 AND ac.voucher_id IS NULL)
-                                OR (rl.voucher_id = ac.voucher_id)) and rl.cleared is true)
-                WHERE ac.cleared IS FALSE
+                                OR (rl.voucher_id = ac.voucher_id)) and rl.cleared IS TRUE)
+                WHERE (ac.cleared IS FALSE OR ac.cleared_on IS NULL)
                         AND ac.chart_id = (select chart_id from cr_report where id = in_report_id)
                 GROUP BY gl.ref, ac.source, ac.transdate,
-                        ac.memo, ac.voucher_id, gl.table
+                        ac.memo, ac.voucher_id, gl.table, ac.entry_id
                 HAVING count(rl.report_id) > 0) a
         LOOP
                 ac_entries := ac_entries || current_row.entries;
         END LOOP;
 
-        UPDATE acc_trans SET cleared = TRUE
-        WHERE entry_id = any(ac_entries);
+        FOREACH ac_entry IN ARRAY ac_entries LOOP
+            UPDATE acc_trans
+               SET cleared = TRUE, cleared_on = clear_time, reconciled_on = now()
+              FROM cr_report_line
+             WHERE ledger_id = ac_entry
+               AND entry_id = ac_entry
+               AND report_id = in_report_id;
+        END LOOP;
```

Comment:
Since the UPDATE doesn't depend on the value of `ac_entry`, why does this need to be a LOOP?

Wouldn't the `UPDATE .... WHERE entry_id = ANY(...)` be more efficient?